### DISCUSSION TITLE
Higher `input_delay` required to prevent lag in updating screen

### DISCUSSION THREAD

**Discussion by PaulVanSchayck** [kovidgoyal/kitty#6493] [Q&A]
First of all, thanks for Kitty, it's a really nice terminal and I love the features.

I had however some trouble setting it up. Initially I experienced the following things:

* Lag for the shell prompt, or the command output to appear after a command. This did not always happen, but about 10% of the commands. The output would be delayed by a second or more.
* Flickering when continuously press enter to flood the screen with shell prompts. The bottom prompt on screen would appear to flicker.
* Partial updates of the screen when PgDwn or PgUp'ing through a large document in `nano`. Sometimes the old page would remain partial visible for one or two seconds, before being updated to the new page. This could be half a page, or only a couple of lines.

I spent a lot of time trying to debug this. Blaming it on shell integration, font, `repaint_delay` or `sync_to_monitor`. Always testing with a `kitty -c NONE` to prevent other settings from interfering.

What worked for me finally was increasing the `input_delay` setting to 20 ms. After that, all above issues disappeared completely. Kitty is super fast now. 

So sharing this story, because I saw no one with similar issues. And I blamed it for a long time on the shell being slow or laggy. And also because I'm wondering if setting a higher `input_delay` is the correct fix, or if it's hiding something else. Does anyone know?

Maybe related, is that I use a relatively high key repetition rate of 50 per second (`xset r rate 200 50`). And this is all on a Linux Pop!OS 22.04 install, with the latest kitty 0.29.1

**Comment by kovidgoyal** [comment]
This will be because the software you are using is drawing too slowly and not using the atomic screen update facilities modern terminals like kitty provide. So kitty will end up rendering partial updates. increasing input delay just causes kitty to wait longer before rendering input it receives giving these programs time to finish drawing.

**Reply by PaulVanSchayck** [reply]
What software I'm using are you referring to? I guess `nano` may be too slow. But it's also the shell (`bash`). I did a quick try with `zsh` and got the same occasional 1+ second delay waiting for the command output to appear, or the prompt to appear after command output. I'm typically using `ls` in my home directory to test, but it happens with many commands

What shell am I supposed to use that are able to use the atomic screen updates?

**Comment by kovidgoyal** [comment]
If you are seeing a shell prompt not being drawn for a second, there is
definitely something wrong with your shell rc files, probably something 
conditioned on TERM

**Reply by PaulVanSchayck** [reply]
I'm using `xterm-kitty` as $TERM as recommended. 

Something wrong with the shell is what I initially also suspected. I tried with `shell` set to `bash --norc --noprofile` and even `env -i bash --norc --noprofile`. With this bare shell I had the same problem. I also tried `zsh`, for which I had nothing extra configured but also same results.

I also found this [very similar report](https://github.com/kovidgoyal/kitty/issues/2485). So I tried setting my keyboard repeat rate to the defaults. But that didn't help.
